language/move-lang/src/parser/syntax.rs

```diff
@@ -579,14 +579,16 @@ fn parse_term<'input>(tokens: &mut Lexer<'input>) -> Result<Exp, Error> {
         Tok::NameValue => {
             // Check if this is a ModuleAccess for a pack or call expression.
             match tokens.lookahead()? {
-                Tok::ColonColon | Tok::LBrace | Tok::LParen => parse_pack_or_call(tokens)?,
+                Tok::ColonColon | Tok::LBrace | Tok::LParen => {
```

Comment:
I think this works, but there are some inconsistencies that bother me. Are the type arguments mandatory for a GenericName AST node? This implementation does not require that. If the ModuleAccess includes a "::" then it will lead to a GenericName. Specifically, compare "Name" vs. "Module::Name" (with no type arguments) -- the first one will parse to a Name and the second to a GenericName with empty type arguments. Using the same Name AST for both would fix that discrepancy, I believe. I think that would be better.

Reply:
I was very confused by the second AST node, I'd like to see it down to one. Makes it more clear what the correct error message is too (you can just say unexpected type arguments or something without referring to a spec)

I'd really like to see it simplified in the parser, but not really sure if its possible (see my previous precedence question)

Reply:
Ok, I 'm doing this now with a single Name node. 

Please note that I was trying to follow a pattern here. (a) there is already GlobalCall and Call nodes, which seems to be quite similar (b) my impression on our first round of changes for specs was that you prefer minimal footprint for specs, and using a joined node definitely has more impact because we need to change all non-spec related handling of the Name node. But it turned out not to be too bad. And I agree with Bob that definitely the logic -- when we created Name and when GenericName -- was subtle, so this is clearer now.